byzcoin/bcadmin/main.go

```diff
@@ -1062,6 +1082,96 @@ func darcShow(c *cli.Context) error {
 	return err
 }
 
+// "cDesc" stands for Change Description. This function allows one to edit the
+// description of a darc.
+func darcCdesc(c *cli.Context) error {
+	bcArg := c.String("bc")
+	if bcArg == "" {
+		return errors.New("--bc flag is required")
+	}
+
+	desc := c.String("desc")
+	if desc == "" {
+		return errors.New("--desc flag is required")
+	}
+	if len(desc) > 1024 {
+		return errors.New("descriptions longer than 1024 characters are not allowed")
+	}
+
+	cfg, cl, err := lib.LoadConfig(bcArg)
+	if err != nil {
+		return err
+	}
+
+	did := c.String("did")
+	if did == "" {
+		did = cfg.AdminDarc.GetIdentityString()
+	}
+
+	d, err := getDarcByString(cl, did)
+	if err != nil {
+		return err
+	}
+
+	d2 := d.Copy()
+	err = d2.EvolveFrom(d)
+	if err != nil {
+		return err
+	}
+
+	d2.Description = []byte(desc)
+	d2Buf, err := d2.ToProto()
+	if err != nil {
+		return err
+	}
+
+	var signer *darc.Signer
+
+	sstr := c.String("sign")
+	if sstr == "" {
+		signer, err = lib.LoadKey(cfg.AdminIdentity)
+	} else {
+		signer, err = lib.LoadKeyFromString(sstr)
+	}
+	if err != nil {
+		return err
+	}
+
+	counters, err := cl.GetSignerCounters(signer.Identity().String())
+
+	invoke := byzcoin.Invoke{
+		ContractID: byzcoin.ContractDarcID,
+		Command:    "evolve_unrestricted",
```

Comment:
This should be "evolve", because we are not changing the rules. See the comments on the top of contract_darc.go.